src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.Batch.cs

```diff
@@ -0,0 +1,43 @@
+        public override int Timeout { get => throw null; set { } }
+        public new Microsoft.Data.SqlClient.SqlConnection Connection { get => throw null; set { } }
+        public new Microsoft.Data.SqlClient.SqlTransaction Transaction { get => throw null; set { } }
+        protected override System.Collections.Generic.IList<System.Data.Common.DbBatchCommand> DbBatchCommands { get => throw null; }
+        protected override System.Data.Common.DbConnection DbConnection { get => throw null; set { } }
+        protected override System.Data.Common.DbTransaction DbTransaction { get => throw null; set { } }
+        public override void Cancel() => throw null;
+        public override int ExecuteNonQuery() => throw null;
+        public override System.Threading.Tasks.Task<int> ExecuteNonQueryAsync(System.Threading.CancellationToken cancellationToken = default) => throw null;
+        public override object ExecuteScalar() => throw null;
+        public override System.Threading.Tasks.Task<object> ExecuteScalarAsync(System.Threading.CancellationToken cancellationToken = default) => throw null;
+        public override void Prepare() => throw null;
+        public override System.Threading.Tasks.Task PrepareAsync(System.Threading.CancellationToken cancellationToken = default) => throw null;
+        protected override System.Data.Common.DbDataReader ExecuteDbDataReader() => throw null;
+        protected override System.Threading.Tasks.Task<System.Data.Common.DbDataReader> ExecuteDbDataReaderAsync(System.Threading.CancellationToken cancellationToken) => throw null;
+        public System.Collections.Generic.List<Microsoft.Data.SqlClient.SqlBatchCommand> Commands { get { throw null; } }
+        public Microsoft.Data.SqlClient.SqlDataReader ExecuteReader() => throw null;
+    }
+    public partial class SqlBatchCommand : System.Data.Common.DbBatchCommand
+    {
+        public SqlBatchCommand() { throw null; }
+        public SqlBatchCommand(string commandText, System.Data.CommandType commandType = System.Data.CommandType.Text, System.Collections.Generic.IEnumerable<Microsoft.Data.SqlClient.SqlParameter> parameters = null) { throw null; }
+        public new Microsoft.Data.SqlClient.SqlParameterCollection Parameters { get { throw null; } }
+        public override string CommandText { get { throw null; } set { } }
+        public override System.Data.CommandType CommandType { get { throw null; } set { } }
+        public override System.Data.CommandBehavior CommandBehavior { get { throw null; } set { } }
+        public override int RecordsAffected { get { throw null; } set { } }
+        protected override System.Data.Common.DbParameterCollection DbParameterCollection => throw new System.NotImplementedException();
+    }
+
+    public sealed partial class SqlException
+    {
+        public Microsoft.Data.SqlClient.SqlBatchCommand SqlBatchCommand { get { throw null; } }
```

Comment:
```suggestion
        public new Microsoft.Data.SqlClient.SqlBatchCommand BatchCommand { get { throw null; } }
```

Cf. https://github.com/npgsql/npgsql/blob/8d6da14034aedb88ec5ad98474a99fa38e670535/src/Npgsql/NpgsqlException.cs#L54

Reply:
I've renamed it but I don't need to use the `new` modifier. That seems to indicate that it's not picking up net6 definitions. Not sure what's going on with the build.

Reply:
fixing up the ref inclusions seems to have worked this loose and it now needs new and has the name suggested.